File: src/bentley_ottmann/event.rs
use traiter::numbers::Parity;

pub(super) type Event = usize;

pub(super) fn is_left_event(event: Event) -> bool {
    event.is_even()
}

pub(super) fn segment_id_to_left_event(segment_id: usize) -> Event {
    segment_id * 2
}

pub(super) fn segment_id_to_right_event(segment_id: usize) -> Event {
    segment_id * 2 + 1
}

File: rene/_exact/bentley_ottmann/event.py
from typing import NewType

Event = NewType('Event', int)


def is_left_event(event: Event) -> bool:
    return event % 2 == 0


def segment_id_to_left_event(segment_id: int) -> Event:
    return Event(2 * segment_id)


def segment_id_to_right_event(segment_id: int) -> Event:
    return Event(2 * segment_id + 1)
